File: Leetcode/190. Reverse Bits/190_Reverse_Bits.py
from typing import List


class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0
        for i in range(32):
            bit = (n >> i) & 1
            result = result | (bit << (31 - i))
        return result


class Main:
    def main(self):
        solution = Solution()
        n = 43261596
        result = solution.reverseBits(n)
        print(result)


if __name__ == "__main__":
    main = Main()
    main.main()

File: Leetcode/190. Reverse Bits/190_Reverse_Bits.java
class Solution {
    public int reverseBits(int n) {
        int result = 0;
        for (int i = 0; i < 32; i++) {
            int bit = (n >> i) & 1;
            result = result | (bit << (31 - i));
        }
        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int n = 43261596;
        System.out.println(solution.reverseBits(n));
    }
}